File: BidWinKoModify/app/src/main/java/com/bidwinko/screens/fragments/BuyBidFragment.java
package com.bidwinko.screens.fragments;

import android.app.Activity;
import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.text.Html;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.bidwinko.R;
import com.bidwinko.screens.activity.ShareEarnActivity;
import com.bidwinko.adapter.BuyBidListAdapter;
import com.bidwinko.API.APIService;
import com.bidwinko.model.BuyBid;
import com.bidwinko.model.BuyBidValue;
import com.bidwinko.utilies.Constants;
import com.bidwinko.API.Retrofit;


import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class BuyBidFragment extends Fragment  {

	View view = null;
	RecyclerView recyclerView;
	ProgressDialog progressDialog;
	ArrayList<BuyBidValue> buybidArraylists = new ArrayList<>();
	private BuyBidListAdapter mAdapter;
	TextView txtBidBalance;
	CardView unlimitedBid;

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
		 view = getActivity().getLayoutInflater().inflate(R.layout.buybid_fragment,null);

		((AppCompatActivity)getActivity()).getSupportActionBar().setTitle((Html.fromHtml("<font color=\"#ffffff\">"+getString(R.string.app_name)+" / " + getString(R.string.buybid) + "</font>")));
		 recyclerView = view.findViewById(R.id.recycler_view);
		 txtBidBalance = view.findViewById(R.id.bidbalance);
		 unlimitedBid = view.findViewById(R.id.card_unlimited_bid);

//		 getBuyBids();

		unlimitedBid.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View view) {
//				Toast.makeText(getActivity(),"Feature Coming Soon..",Toast.LENGTH_SHORT).show();
				Intent intent = new Intent(getActivity(),ShareEarnActivity.class);
				startActivity(intent);
			}
		});

		return view;
	}

	public static BuyBidFragment newInstance() {
		return new BuyBidFragment();
	}

//	private void getBuyBids() {
//
//		APIService apiService = Retrofit.getClient().create(APIService.class);
//		Call<BuyBid> call = apiService.getbuyBids(Constants.getSharedPreferenceInt(getActivity(),"userId",0),
//				Constants.getSharedPreferenceString(getActivity(),"securitytoken",""),
//				Constants.getSharedPreferenceString(getActivity(),"versionName",""),
//				Constants.getSharedPreferenceInt(getActivity(),"versionCode",0),
//				Constants.getSharedPreferenceString(getActivity(),"userFrom",""));
//
//		if(!((Activity) getActivity()).isFinishing()) {
//			progressDialog = new ProgressDialog(getActivity());
//			progressDialog.setMessage(getString(R.string.loadingwait));
//			progressDialog.show();
//			progressDialog.setCancelable(false);
//		}
//
//		call.enqueue(new Callback<BuyBid>() {
//			@Override
//			public void onResponse(Call<BuyBid>call, Response<BuyBid> response) {
//				dismissProgressDialog();
//
//				if(response!=null){
//					if(response.isSuccessful()){
//						if(response.body().getStatus()==200){
////							txtBidBalance.setText(""+response.body().getBidBal());
//							buybidArraylists = response.body().getBuyBidValues();
//							String bidremain = response.body().getBidBalance();
//
////							Log.e("bidremain", "onResponse:pd_BB "+bidremain );
//							Constants.setSharedPreferenceString(getActivity(),"bidBalance",bidremain);
//							txtBidBalance.setText(""+Constants.getSharedPreferenceString(getActivity(),"bidBalance",""));
//							mAdapter = new BuyBidListAdapter(response,buybidArraylists,getActivity());
//							RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(getActivity());
//							recyclerView.setLayoutManager(mLayoutManager);
//							recyclerView.setItemAnimator(new DefaultItemAnimator());
//							recyclerView.setAdapter(mAdapter);
//						}else{
//							Toast.makeText(getActivity(),getString(R.string.systemmessage)+response.body().getMessage(),Toast.LENGTH_SHORT).show();
//						}
//
//					}
//				}
//				else{
//					Toast.makeText(getActivity(),getString(R.string.systemmessage)+response.errorBody(),Toast.LENGTH_SHORT).show();
//				}
//
//
//			}
//
//			@Override
//			public void onFailure(Call<BuyBid>call, Throwable t) {
//				// Log error here since request failed
//				Log.e("response", t.toString());
//			}
//		});
//
//
//	}

	private void dismissProgressDialog() {
		if (progressDialog != null && progressDialog.isShowing()) {
			progressDialog.dismiss();
		}
	}

	@Override
	public void onDestroy() {
		dismissProgressDialog();
		super.onDestroy();
	}

	@Override
	public void onPause() {
		dismissProgressDialog();
		super.onPause();
	}


	@Override
	public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
		super.onActivityResult(requestCode, resultCode, data);
		if(requestCode==103){
			getActivity().finish();
			startActivity(getActivity().getIntent());
		}
	}
}

File: BidWinKoModify/app/src/main/java/com/bidwinko/screens/fragments/BuyBidFragment.kt
package com.bidwinko.screens.fragments

import android.animation.Animator
import android.animation.Animator.AnimatorListener
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.recyclerview.widget.RecyclerView
import com.bidwinko.R
import com.bidwinko.adapter.BuyBidPlanAdapter
import com.bidwinko.databinding.BuybidFragmentBinding
import com.bidwinko.model.RequestModels.CommonRequest
import com.bidwinko.model.ResponseModels.BidPlan
import com.bidwinko.screens.activity.ReferActivity
import com.bidwinko.screens.activity.ShareEarnActivity
import com.bidwinko.utilies.Constants
import com.bidwinko.utilies.SessionManager
import com.bidwinko.viewModel.mainViewModel

class BuyBidFragment : Fragment() {

    lateinit var binding: BuybidFragmentBinding

    var recyclerView: RecyclerView? = null
    private var mAdapter: BuyBidPlanAdapter? = null
    var txtBidBalance: TextView? = null
    var unlimitedBid: CardView? = null
    private lateinit var const: FragmentActivity

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is FragmentActivity) {
            const = context
        }
    }
    
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View {
        
        binding = BuybidFragmentBinding.inflate(layoutInflater)
        const.findViewById<TextView>(R.id.title).text = getString(R.string.buybid)
        recyclerView = binding.bidRecyclerView
        txtBidBalance = binding.bidbalance
        unlimitedBid = binding.cardUnlimitedBid
        binding.shimmer.root.startShimmer()
        binding.anim.let {
            it.playAnimation()
            it.addAnimatorListener(object : AnimatorListener{
                override fun onAnimationStart(animation: Animator) {

                }

                override fun onAnimationEnd(animation: Animator) {
                    it.playAnimation()
                }

                override fun onAnimationCancel(animation: Animator) {

                }

                override fun onAnimationRepeat(animation: Animator) {

                }

            })
        }
        binding.bidbalance.text = SessionManager(const).GetValue(Constants.TOTAL_BIDS)
        getBuyBids()
        unlimitedBid!!.setOnClickListener(View.OnClickListener { //				Toast.makeText(getActivity(),"Feature Coming Soon..",Toast.LENGTH_SHORT).show();
            val intent = Intent(activity, ReferActivity::class.java)
            startActivity(intent)
        })
        return binding.root
    }

    private fun getBuyBids() {
        val request = CommonRequest(
            userId = SessionManager(const).GetValue(Constants.USER_ID).toString(),
            securityToken = SessionManager(const).GetValue(Constants.SECURITY_TOKEN),
            versionName = SessionManager(const).GetValue(Constants.VERSION_NAME),
            versionCode = SessionManager(const).GetValue(Constants.VERSION_CODE)
        )
        mainViewModel(const).GetBidsPacakage(request).observe(const) {
            if (it.status == 200) {
                if (it.bidPlans.size > 0) {
                    mAdapter =
                        BuyBidPlanAdapter(it.bidPlans as ArrayList<BidPlan>, const)
                    binding.bidRecyclerView.setAdapter(mAdapter)
                    binding.bidRecyclerView.visibility = View.VISIBLE
                    binding.shimmer.root.stopShimmer()
                    binding.shimmer.root.visibility = View.GONE
                } else {
                    binding.shimmer.root.stopShimmer()
                    binding.shimmer.root.visibility = View.GONE
                    binding.nodatafound.visibility = View.VISIBLE
                    binding.animm.addAnimatorListener(object : Animator.AnimatorListener {
                        override fun onAnimationStart(animation: Animator) {
                        }

                        override fun onAnimationEnd(animation: Animator) {
                            binding.animm.playAnimation()
                        }

                        override fun onAnimationCancel(animation: Animator) {
                        }

                        override fun onAnimationRepeat(animation: Animator) {
                        }

                    })
                }
            }
        }
    }


    override fun onDestroy() {

        binding.shimmer.root.stopShimmer()
        super.onDestroy()
    }

    override fun onPause() {

        binding.shimmer.root.stopShimmer()
        super.onPause()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == 103) {
            const.finish()
            startActivity(const.intent)
        }
    }

    companion object {
        fun newInstance(): BuyBidFragment {
            return BuyBidFragment()
        }
    }
}
